### PR TITLE
test: Remove circular symlinks from testdata

### PR DESCRIPTION
**Pull request by jannfis**
This removes circular symlinks from testdata of repo server, and instead creates the links at runtime during the tests.

This fixes build breakage on certain infrastructures.

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: reposerver/repository/repository_test.go [hunk 1]
```diff
@@ -133,6 +133,31 @@ func newServiceWithCommitSHA(root, revision string) *Service {
...
+	if err != nil {
+		return err
+	}
+	if workingDir != "" {
+		err = os.Chdir(workingDir)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			if err := os.Chdir(oldWorkingDir); err != nil {
```
Comment (crenshaw-dev) [inline reposerver/repository/repository_test.go:149]: Is it possible we're gonna run into race conditions when other tests assume a certain working directory?

File: reposerver/repository/repository_test.go [hunk 2]
```diff
@@ -133,6 +133,31 @@ func newServiceWithCommitSHA(root, revision string) *Service {
...
+				t.Fatal(err.Error())
+			}
+		}()
+	}
+	err = os.Symlink(destName, linkName)
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (crenshaw-dev) [inline reposerver/repository/repository_test.go:158]: Could just do `return os.Symlink(destName, linkName)`.

**Review by crenshaw-dev — APPROVED** [review]
lgtm besides one nit.  :-)

**Comment by jannfis** [comment]
Cherry-picking this into release-2.4

**Pull request merged**